### PR TITLE
feat: fetch predictions from S3 data lake with Athena

### PR DESCRIPTION
**Pull request by kartikey-vyas**
WIP

This PR includes the following changes:
- remove `pandera` and model-based unit tests; not enough wide usage of `pandera`, it conflicts with newer `pydantic` versions and there are other ways to validate data in base python
- create `PredictionFetcher` class to encapsulate all prediction fetching logic
    - class was chosen as there is a lot of re-used state information (AWS resource names particularly)
    - capturing state could be useful down the line as we collect more metadata
    - metadata for the user/reader side can be kept within this class if we want

To do:
- define a table schema for `predictions` and `results` explicitly
- notebook which uses the interface as an example
- ~implementation as lambda~

Follow up PRs:

**feat: metadata in data lake** #27
- set up `metadata` table on S3 + Athena
- refactor `Metadata` capture functions to point to data lake

**feat: decommission DynamoDB** #28 
- remove from IaC
- remove related functions and data models

### CODE REVIEW

File: lambda/fetch_predictions.py
Comment (nicholas-kondal) [inline lambda/fetch_predictions.py:1]: How do you see this Lambda being called from the CLI? And would there be another Lambda in the works that generates a pre-signed URL to upload the inputs first?

Right now I've just assumed there would be URLs for each Lambda (see [this draft PR](https://github.com/ersilia-os/ersilia/pull/1190/files#diff-c07c157ec3f6da3baee1cd9cccd875b304da30507c10271d386b300511b6ca21R16-R32)), but maybe you had something else in mind

Comment (kartikey-vyas) [reply lambda/fetch_predictions.py:1]: There will be another one which does the pre-signed url stuff, atm im just assuming we have an input file in the right place

File: lambda/fetch_predictions.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    """Placeholder lambda function which will call main
+
+    Args:
+        event (dict): _description_
+        context (dict): _description_
+
+    Returns:
+        dict: _description_
+    """
+    return {"event": len(event), "context": len(context)}
```
Comment (nicholas-kondal) [inline lambda/fetch_predictions.py:58]: @kartikey-vyas I wonder if a link to the S3 folder containing the outputs for a particular request should be returned after this Lambda runs successfully... sending the predictions as a response over the network might not be suitable for large input files

**Comment by kartikey-vyas** [comment]
This piece of work is sufficiently far enough along for functionality. Further work will be to incorporate metadata, then to use as a lambda.

**Pull request merged**